napari/utils/settings/_defaults.py

```diff
@@ -175,17 +175,70 @@ class QtBindingChoice(str, Enum):
     pyqt5 = 'pyqt5'
 
 
+def yaml_config_settings_source(settings: BaseSettings) -> Dict[str, Any]:
+    """Load and validate settings coming from configuration file."""
+    yaml_settings = {}
+    model_class = settings.__class__
+
+    # This is set by the SettingsManager
+    loaded_data = getattr(settings, "_LOADED_DATA", {})
+
+    validate = getattr(settings, "_IGNORE_YAML_SOURCE", False)
+    if not validate and loaded_data:
+        # This variable prevents recursion when using the model for validation
+        model_class._IGNORE_YAML_SOURCE = True
+
+        # Get defaults from the schema
+        model_schema = model_class.schema()
+        section = model_schema["section"]
+        default_properties = model_schema.get("properties", {})
+        yaml_settings = loaded_data.get(section, {}).copy()
+
+        try:
+            model_class(**yaml_settings)
+        except KeyError:
```

Comment:
what causes the potential KeyError happen here?

Reply:
It came from the original code I copied 
https://github.com/napari/napari/blob/master/napari/utils/settings/_manager.py#L147

Indeed it is now not needed. Good catch 🎸 !

Reply:
Fixed!